reddit_service_websockets/dispatcher.py

```diff
@@ -22,6 +46,14 @@ def __init__(self, metrics):
     def on_message_received(self, namespace, message):
         consumers = self.consumers.get(namespace, [])
 
+        # Compress the message
+        compressed = make_compressed_frame(message, COMPRESSOR)
+        message = Message(
+            compressed=compressed,
+            raw=message,
+            percent_compressed=1 - float(len(compressed))/float(len(message)),
```

Comment:
Yeah, my thinking was just that if we do it downstream, the calculation will have to happen once for every time the message is sent out, since there's a pump dispatcher for every client.  I indeed don't like it here either (which is why I put the `max(0, [...])` math elsewhere, but now that I think about it, that's the same problem.  Do you think it's worth it to repeat the calculation for every client or make the compromise on what the Message object is caring about?